entity-framework/core/modeling/relationships/conventions.md

```diff
@@ -0,0 +1,450 @@
+        }
+-->
+[!code-csharp[OneToManySingleRelationship](../../../../samples/core/Modeling/Relationships/RelationshipConventions.cs?name=OneToManySingleRelationship)]
+
+A single, one-to-one relationship is discovered between `Blog` and `Author` is discovered by pairing the `Blog.Author` and `Author.Blog` navigations:
+
+<!--
+        public class Blog
+        {
+            public int Id { get; set; }
+            public Author? Author { get; set; }
+        }
+
+        public class Author
+        {
+            public int Id { get; set; }
+            public int? BlogId { get; set; }
+            public Blog? Blog { get; set; }
+        }
+-->
+[!code-csharp[OneToOneSingleRelationship](../../../../samples/core/Modeling/Relationships/RelationshipConventions.cs?name=OneToOneSingleRelationship)]
+
+A single, many-to-many relationship is discovered between `Post` and `Tag` is discovered by pairing the `Post.Tags` and `Tag.Posts` navigations:
+
+<!--
+        public class Post
+        {
+            public int Id { get; set; }
+            public ICollection<Tag> Tags { get; } = new List<Tag>();
+        }
+
+        public class Tag
+        {
+            public int Id { get; set; }
+            public ICollection<Post> Posts { get; } = new List<Post>();
+        }
+-->
+[!code-csharp[ManyToManySingleRelationship](../../../../samples/core/Modeling/Relationships/RelationshipConventions.cs?name=ManyToManySingleRelationship)]
+
+> [!NOTE]
+> This pairing of navigations may be incorrect if the two navigations represent two, different, unidirectional relationships. In this case, the two relationships must be configured explicitly.
+
+Pairing of relationships only works when there is a single relationship between two types. Multiple relationships between two types must be configured explicitly.
+
+> [!NOTE]
+> The descriptions here are in terms of relationships between two different types. However, it is possible for the same type to be on both ends of a relationship, and therefore for a single type to have two navigations both paired with each other. This is called a self-referencing relationship.
+
+## Discovering foreign key properties
+
+Once the navigations for a relationship have either been discovered or configured explicitly, then these navigations are used to discover appropriate foreign key properties for the relationship. A property is discovered as a foreign key when:
+
+- The property type is compatible with the primary or alternate key on the principal entity type.
+  - Types are compatible if they are the same, or if the foreign key property type is a nullable version of the primary or alternate key property type.
+- The property name matches one of the naming conventions for a foreign key property. The naming conventions are:
+  - `<navigation property name><principal key property name>`
+  - `<navigation property name>Id`
+  - `<principal entity type name><principal key property name>`
+  - `<principal entity type name>Id`
+
```

Comment:
We also try to use the dependent PK as the FK if the dependent end was configured explicitly